### PR TITLE
[UI] start with the breadcrumb component

### PR DESCRIPTION
**Pull request by sstraatemans**
## Reason:

Closes [issue link]

## Check off the following:

- [ ] I have reviewed my changes and run the appropriate tests.
- [ ] I have have run `rush change` to add the appropriate change logs.
- [ ] I have added/edited docs.
- [ ] I have added tutorials.
- [ ] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { ProcuctIcons } from '../ProductIcons';
```
Comment (isaozler) [inline packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.stories.tsx:1]: typo

**Review by isaozler — APPROVED** [review]

**Review by ash-vd — APPROVED** [review]

File: packages/libs/react-components/src/components/Breadcrumbs/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  a: {
+    color: '$neutral4',
+    textDecoration: 'none',
+    '&:hover': {
+      textDecoration: 'underline',
+    },
+  },
+  span: {
+    marginTop: 'calc($1 /2)',
```
Comment (eileenmguo) [inline packages/libs/react-components/src/components/Breadcrumbs/styles.tsx:46]: does this work? I would expect it to have to be 'calc($space$1 / 2)'

**Review by eileenmguo — CHANGES_REQUESTED** [review]

File: packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export interface IBreadcrumbs {
+  children?: FunctionComponentElement<IBreadcrumbItem>[];
+  icon?: typeof ProductIcons[keyof typeof ProductIcons];
+}
+
+export const Breadcrumbs: FC<IBreadcrumbs> = ({ children, icon }) => {
+  return (
+    <StyledBreadcrumbs>
+      {React.Children.map(children, (child, idx) => {
```
[the remark this answers is not in the document]
Comment (sstraatemans) [reply packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx:16]: that does not exist unfortunatly

File: packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        if (
+          child === undefined ||
+          child.type.displayName !== BreadcrumbItem.displayName
```
[the remark this answers is not in the document]
Comment (sstraatemans) [reply packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx:19]: makes sense.

File: packages/libs/react-components/src/components/Breadcrumbs/styles.tsx (hunk 2 above)
Comment (sstraatemans) [reply packages/libs/react-components/src/components/Breadcrumbs/styles.tsx:46]: the weird thing is $spaces$1 does not work here

File: packages/libs/react-components/src/components/Breadcrumbs/BreadcrumbItem.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  icon?: typeof ProductIcons[keyof typeof ProductIcons];
+}
+
+export const BreadcrumbItem: FC<IBreadcrumbItem> = ({ children, icon }) => {
+  const Icon = icon;
+  return (
+    <StyledBreadcrumbItem>
+      {React.Children.map(children, (child) => {
+        if (child === undefined) return;
+        if (!React.isValidElement(child)) return child;
```
Comment (eileenmguo) [inline packages/libs/react-components/src/components/Breadcrumbs/BreadcrumbItem.tsx:18]: What is this case for?

File: packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+export const Breadcrumbs: FC<IBreadcrumbs> = ({ children, icon }) => {
+  return (
+    <StyledBreadcrumbs>
+      {React.Children.map(children, (child, idx) => {
+        if (child === undefined || child.type !== BreadcrumbItem) {
+          throw new Error(
+            `${child?.type} is not a valid child for Breadcrumbs`,
+          );
+        }
+        console.log(child.type !== BreadcrumbItem);
```
Comment (eileenmguo) [inline packages/libs/react-components/src/components/Breadcrumbs/Breadcrumbs.tsx:22]: 👀

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**